oshmem/mca/spml/ucx/spml_ucx.h

```diff
@@ -62,7 +65,8 @@ typedef struct spml_ucx_cached_mkey spml_ucx_cached_mkey_t;
 
 struct ucp_peer {
     ucp_ep_h                 ucp_conn;
-    spml_ucx_cached_mkey_t   mkeys[MCA_MEMHEAP_MAX_SEGMENTS];
+    spml_ucx_cached_mkey_t **mkeys;
```

Comment:
why do you use ```spml_ucx_cached_mkey_t **mkeys``` - indirect array instead of ```spml_ucx_cached_mkey_t *mkeys```? does it provide better memory consumption?

Reply:
We sorted it out with @hoopoepg offline.
Reason is that we have external pointer in the current implementation referencing some elements. Using array of objects will break the reference at realloc.